OpenRA.Mods.Common/Traits/World/SpawnMapActors.cs

```diff
@@ -26,6 +26,9 @@ public class SpawnMapActors : IWorldLoaded
 
 		public void WorldLoaded(World world, WorldRenderer wr)
 		{
+			var iPreventMapSpawns = world.WorldActor.TraitsImplementing<IPreventMapSpawn>()
+				.Concat(world.WorldActor.Owner.PlayerActor.TraitsImplementing<IPreventMapSpawn>());
```

Comment:
This will need a `ToList` to turn it into a concrete list, otherwise it will reevaluate the enumerator and repeat the lookups each time!

Reply:
Style nit: I don't think we've used `iWhatever` anywhere else, so how about calling this `preventMapSpawn`?